corehq/middleware.py

```diff
@@ -97,16 +101,39 @@ def process_response(self, request, response):
 
 
 class LogLongRequestMiddleware(MiddlewareMixin):
+    """Report requests that violate the timing threshold configured for the view.
+
+    Use `corehq.util.timer.set_request_duration_reporting_threshold` to override the
+    default threshold for specific views.
+    """
+    DEFAULT_THRESHOLD = timedelta(minutes=10).total_seconds()  # 10 minutes
 
     def process_request(self, request):
         request._profile_starttime = datetime.datetime.utcnow()
 
+    def process_view(self, request, view_fn, view_args, view_kwargs):
+        view_func = get_view_func(view_fn, view_kwargs)
+        reporting_threshold = getattr(view_func, DURATION_REPORTING_THRESHOLD, self.DEFAULT_THRESHOLD)
+        setattr(request, DURATION_REPORTING_THRESHOLD, reporting_threshold)
+
     def process_response(self, request, response):
+        request_timer = getattr(response, 'request_timer', None)
+        if request_timer:
+            for sub in request_timer.to_list(exclude_root=True):
+                add_breadcrumb(
+                    category="timing",
+                    message=f"{sub.name}: {sub.duration:0.3f}",
+                    level="info",
+                )
+
         if hasattr(request, '_profile_starttime'):
             duration = datetime.datetime.utcnow() - request._profile_starttime
-            if duration > datetime.timedelta(minutes=10):
-                notify_exception(request, "Request took a very long time.", details={
+            threshold = getattr(request, DURATION_REPORTING_THRESHOLD, self.DEFAULT_THRESHOLD)
+            if duration.total_seconds() > threshold:
```

Comment:
If `DURATION_REPORTING_THRESHOLD` and `DEFAULT_THRESHOLD` held `timedelta` objects `threshold` could be compared directly to `duration`.
```suggestion
            threshold = getattr(request, DURATION_REPORTING_THRESHOLD, self.DEFAULT_THRESHOLD)
            if duration > threshold:
```